playground/hmr/__tests__/hmr.spec.ts

```diff
@@ -688,19 +701,16 @@ if (!isBuild) {
 
     await page.goto(viteTestUrl + '/missing-import/index.html')
 
-    browserLogs.length = 0
-    expect(browserLogs).toMatchObject([])
-
-    editFile(file, (code) => code.replace(importCode, unImportCode))
-
-    await page.waitForNavigation({ timeout })
-    expect(browserLogs.some((msg) => msg.match('missing test'))).toBe(true)
-    browserLogs.length = 0
-
-    editFile(file, (code) => code.replace(unImportCode, importCode))
-
-    await page.waitForNavigation({ timeout })
-    expect(browserLogs.some((msg) => msg.includes('500'))).toBe(true)
-    browserLogs.length = 0
+    await untilBrowserLogAfter(async () => {
+      const navigationPromise = page.waitForNavigation({ timeout })
+      editFile(file, (code) => code.replace(importCode, unImportCode))
+      await navigationPromise
```

Comment:
Navigation could happen between `editFile` and `page.waitForNavigation({ timeout })`. So if the execution order were:

1. `editFile`
2. navigation
3. `page.waitForNavigation`

The promise of `page.waitForNavigation` will never fulfill.

Reply:
Maybe this fail was caused by this.
https://github.com/vitejs/vite/actions/runs/3420151918/jobs/5714504703#step:12:102